crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -318,6 +322,16 @@ fn impl_struct(
                 #partial_eq_fn
             }
         }
+
+        impl #impl_generics #bevy_reflect_path::Typed for #struct_name #ty_generics #where_clause {
+            fn type_info() -> #bevy_reflect_path::TypeInfo {
+                let fields: [#bevy_reflect_path::NamedField; #field_count] = [
```

Comment:
Pretty sure can't be done due to `const` interaction with generic types, but I have to ask anyway: can we make this `const`? It would eliminate runtime allocations for this.

Reply:
Within a macro, we can probably branch on whether or not the type has generics.

Reply:
Do you mean make the `type_info()` method `const`? I'm not sure that's allowable on trait methods.

Reply:
> I'm not sure that's allowable on trait methods.

Correct, we can't make trait methods constant, so this idea won't work.

Reply:
I mean we could maybe add an associated `const TYPE_INFO: TypeInfo`. But even if that did work, we can't make `TypeInfo` constant due to the issue with `TypeId::of::<T>()` not being `const`.